src/System.Management.Automation/engine/Utils.cs

```diff
@@ -238,20 +238,16 @@ internal static string GetApplicationBaseFromRegistry(string shellId)
         }
 #endif
 
-        /// <summary>
-        /// Gets the application base for current monad version
-        /// </summary>
-        /// <returns>
-        /// applicationbase path for current monad version installation
-        /// </returns>
-        /// <exception cref="SecurityException">
-        /// if caller doesn't have permission to read the key
-        /// </exception>
+        internal static string DefaultPowerShellAppBase { get; } = GetApplicationBase(DefaultPowerShellShellID);
+
         internal static string GetApplicationBase(string shellId)
         {
 #if CORECLR
-            // Use the location of SMA.dll as the application base
-            // Assembly.GetEntryAssembly and GAC are not in CoreCLR.
+            if (DefaultPowerShellAppBase != null)
+            {
+                return DefaultPowerShellAppBase;
+            }
```

Comment:
This is a bit confusing -- looks like a recursion would happen even though it's not. I suggest removing this if block.